### PR TITLE
KAFKA-7029: Update ReplicaVerificationTool to use Java Consumer

### PR DESCRIPTION
**Pull request by omkreddy**
### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@ijuma  Pls take a look. I have verified the unit test and System test related this tool.  Also done manual testing.

File: core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala [hunk 1]
```diff
@@ -87,7 +98,7 @@ object ReplicaVerificationTool extends Logging {
                          .defaultsTo(".*")
     val initialOffsetTimeOpt = parser.accepts("time", "Timestamp for getting the initial offsets.")
                            .withRequiredArg
-                           .describedAs("timestamp/-1(latest)/-2(earliest)")
+                           .describedAs("-1(latest)/-2(earliest)")
```
Comment (omkreddy) [inline core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala]: To simply code, I have removed the support to start from a particular timestamp offset. Not sure any one using this. If required I will add.

Comment (ijuma) [reply core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala]: If it's not too complicated, we should probably just add it.

Comment (omkreddy) [reply core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala]: Updated the code.

File: core/src/test/scala/kafka/tools/ReplicaVerificationToolTest.scala [hunk 2]
```diff
@@ -55,7 +55,7 @@ class ReplicaVerificationToolTest {
 
     // If you change this assertion, you should verify that the replica_verification_test.py system test still passes
     assertTrue(s"Max lag information should be in output: `$output`",
-      output.endsWith(": max lag is 10 for partition a-0 at offset 10 among 3 partitions"))
+      output.endsWith(": max lag is 10 for partition a-1 at offset 10 among 3 partitions"))
```
Comment (omkreddy) [inline core/src/test/scala/kafka/tools/ReplicaVerificationToolTest.scala:58]: With TopicPartition changes, expectedReplicasPerTopicAndPartition  Map iteration order changed. Hence this change. System test is passing.

**Review by ijuma — APPROVED** [review]
I pushed a few more tweaks, LGTM now. Does it look OK to you?

**Comment by ijuma** [comment]
One more comment, there is direct usage of NetworkClient here which is not ideal. However, we can look at cleaning it up in the future. We should also consider whether this tool is useful enough for us to keep it. If not, we should deprecate it so that we can remove it in the next major release.

**Comment by omkreddy** [comment]
LGTM.

currently to send fetch request directly to a replica node, we need to access NetworkClient. maybe we should have small util class to send/receive request/responses for testing purpose.

This tool looks useful. maybe we can deprecate, if we are planning to write advanced tests in the presence of faults  (using trogdor framework) in future.

**Comment by ijuma** [comment]
Yes, I understand that there's no other way to write it at the moment.

**Comment by ijuma** [comment]
Both Jenkins jobs passed, the failure for one of them was when communicating with GitHub at the end. Merging to trunk and 2.0.

**Pull request merged**